codec/legacy_global/codec.go

```diff
@@ -0,0 +1,19 @@
+package legacy_global
```

Comment:
can we just call this pkg `legacy` and add a small `doc.go`?

Reply:
Sure